### PR TITLE
Add RPM checksums in purls

### PR DESCRIPTION
**Pull request by mprpic**

### CODE REVIEW

File: sbom/examples/rpm/from-koji.py [hunk 1]
```diff
@@ -7,10 +7,11 @@
...
 from tempfile import TemporaryDirectory
 
+# Script requires these RPMs: brewkoji, rpmdevtools, rpm-build
+# Run with: ./from-koji.py brew <NVR>
+
 source_re = re.compile(r"^(Source\d+):\s*((.*/)?(.*))$")
-tarball_re = re.compile(
-    r"^([^0-9]*)-([0-9a-f\.-]*)[\.-][a-z]"
-)  # Obviously not universal
+tarball_re = re.compile(r"^([^0-9]*)-([0-9a-f\.-]*)[\.-][a-z]")  # Obviously not universal
```
Comment (twaugh) [inline sbom/examples/rpm/from-koji.py:14]: Looks like you (or your editor) reformatted this, but not in the style that `black` does?

File: sbom/examples/rpm/from-koji.py [hunk 2]
```diff
@@ -313,13 +311,13 @@ def handle_srpm(filename, name):
             {
                 "referenceCategory": "PACKAGE-MANAGER",
                 "referenceType": "purl",
-                "referenceLocator": f"pkg:rpm/redhat/{name}@{version}-{release}?arch={arch}",
+                "referenceLocator": f"pkg:rpm/redhat/{name}@{version}-{release}?arch={arch}&checksum={digest}",
```
Comment (twaugh) [inline sbom/examples/rpm/from-koji.py:314]:
```suggestion
                "referenceLocator": f"pkg:rpm/redhat/{name}@{version}-{release}?arch={arch}&checksum=sha256:{digest}",
```

File: sbom/examples/rpm/from-koji.py (hunk 1 above)
Comment (mprpic) [reply sbom/examples/rpm/from-koji.py:14]: Ah, I have my `black` configured for 100-char line limit globally, so it changed this line while I saved the file. Do you want me to revert it?

Comment (mprpic) [reply sbom/examples/rpm/from-koji.py:14]: I'm going to add some linting/formatting harness in a separate PR so we get consistency in all scripts.

Comment (mprpic) [reply sbom/examples/rpm/from-koji.py:14]: Done in https://github.com/RedHatProductSecurity/security-data-guidelines/pull/16

**Pull request merged**